packages/ui-components/src/Badge/Badge.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FunctionComponent } from "react";
+import classNames from "classnames/bind";
+
+import styles from "./Badge.module.scss";
+
+type BadgeCase = "none" | "uppercase";
+type BadgeIntent = "neutral" | "success" | "warning" | "danger";
+
+export interface BadgeProps {
+  content?: string;
+  intent?: BadgeIntent;
+  transformCase?: BadgeCase;
+}
+
+const cx = classNames.bind(styles);
+
+const Badge: FunctionComponent<BadgeProps> = ({
+  content,
+  intent = "neutral",
+  transformCase = "uppercase",
```

Comment:
@nathanstilwell , it is possible to assign default properties with `defaultProps` static method
```javascript
const Badge: FunctionComponent<BadgeProps> = () => { ... }

Badge.defaultProps: Partial<BadgeProps> = {
  intent: "neutral",
  transformCase: "uppercase",
}
```

Reply:
Great question, @koorosh! We decided against using `defaultProps` since it is will be [deprecated for function components](https://github.com/reactjs/rfcs/pull/107).